pkgs/by-name/on/onedriver/package.nix

```diff
@@ -40,13 +40,18 @@ buildGoModule {
     install -Dm644 ./pkg/resources/onedriver-128.png $out/share/icons/onedriver/onedriver-128.png
 
     install -Dm644 ./pkg/resources/onedriver.desktop $out/share/applications/onedriver.desktop
+    install -Dm644 ./pkg/resources/onedriver@.service $out/lib/systemd/user/onedriver@.service
 
     mkdir -p $out/share/man/man1
     installManPage ./pkg/resources/onedriver.1
 
     substituteInPlace $out/share/applications/onedriver.desktop \
       --replace "/usr/bin/onedriver-launcher" "$out/bin/onedriver-launcher" \
       --replace "/usr/share/icons" "$out/share/icons"
+
+    substituteInPlace $out/lib/systemd/user/onedriver@.service \
+      --replace "/usr/bin/onedriver" "$out/bin/onedriver" \
+      --replace "/usr/bin/fusermount" "${fuse}/bin/fusermount"
```

Comment:
I don't have an account to test this, but I am afraid this is incorrect: fusermount must be setuid to work, and only /run/wrappers/bin/fusermount is, not ${fuse}/bin/fusermount. So I expect this to fail.

Reply:
It is not failing on my side, I have been testing locally for some days now and I can mount and unmount. Probably it's because the `fusermount` is not launched by the user but by systemd?

Reply:
> Probably it's because the fusermount is not launched by the user but by systemd?

no because it's a user service, it runs as your user.

Reply:
In the screenshot below you can see the `onedriver@.service` file I am currently using, the Launcher UI shows two drives mounted, the one I am using for testing is `OD`, added using the UI and correctly mounted after the authentification process.

![Screenshot from 2024-03-03 17-15-28](https://github.com/NixOS/nixpkgs/assets/585424/0836de9b-b301-42ec-b062-98cc8df6417e)



In the second screenshot here below I deactivated the OneDrive mount `OD` using the UI (hence it is using the `fusermount` defined in the `onedriver@.service` file) and as you can see it gets unmounted properly (it disappears from Nautilus, it is no longer shown in the `mount` list, ...). My guess is that since it's passing through systemd it does not need the suid. Indeed if I try to launch the `fusermount` bin in `${fuse}/bin/fusermount` I have a permission denied (and I have to pass from the wrapper as you mentioned).


![Screenshot from 2024-03-03 17-15-52](https://github.com/NixOS/nixpkgs/assets/585424/f0d3b576-af75-4501-b62f-6839a7bda793)

Reply:
I modified the derivation and used `/run/wrappers/bin/fusermount` instead of `${fuse}/bin/fusermount` as you suggested and it is still working, so I guess it is safer to follow your path than mine! Today I learned something new, thanks!

Reply:
Oh my bad ! I read the systemd file, and fusermount is only used to unmount, which is not a privileged operation. That solves the mystery. You can go back to your original solution if you prefer.

Reply:
My bad I was wrong. It *is* privileged.

Reply:
One explanation I can imagine is the following:
- the executable always unmounts itself cleanly
- ExecStartStop is only there for unclean shutdown

To test this theory, you can go back to the previous state, kill the service, and have a look at `systemctl status --user`, it should tell you whether fusermount succeeded. (or look at the logs in `journalctl --user -u theunit.service`)

Reply:
> One explanation I can imagine is the following:
> 
> * the executable always unmounts itself cleanly
> * ExecStartStop is only there for unclean shutdown
> 
> To test this theory, you can go back to the previous state, kill the service, and have a look at `systemctl status --user`, it should tell you whether fusermount succeeded. (or look at the logs in `journalctl --user -u theunit.service`)

Mystery solved, you were right: the executable unmounts itself cleanly when using the UI (it does not go through fusermount) and the fusermount is there only for unclean shutdown. Which basically means that if I kill the service abruptly, the drive will stay in an unclean state and therefore become unusable. Using the wrapper everything works properly.

I won't revert my commit then, since this is the behavior we want! Thanks a lot for your help and for your investigation on this, I learned something new :)